### PR TITLE
Progress also writes messaging to output channel

### PR DESCRIPTION
**Pull request by bryceitoc9**
## Description

Progress messages and initial title are written to the output channel as well as to the Progress UI element.

## Motivation and Context

Recommended by UX review

## Testing
Messages appear correctly in both output and progress

## Screenshots (if appropriate)
image

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/vscode/window.ts [hunk 1]
```diff
@@ -136,3 +142,15 @@ class DefaultWindow implements Window {
...
+/**
+ * Outputs progress message to an output channel as well as reporting progress through normal means.
+ * @param progress vscode.Progress object generated through DefaultWindow
+ * @param entry ProgressEntry to output to output channel and progress
+ */
+export function report(progress: vscode.Progress<ProgressEntry>, entry: ProgressEntry) {
+    if (entry.message) {
+        ext.outputChannel.appendLine(entry.message)
+    }
+    progress.report(entry)
```
Comment (justinmk3) [inline src/shared/vscode/window.ts:155]: `withProgress()` already has access to `progress`. Why force all the callers to import this module function? Would it make sense to do it inside `withProgress()` instead?

Comment (bryceitoc9) [reply src/shared/vscode/window.ts:155]: I'll see what I can do. I'd rather not muck up the return object but it would make it neater on the other end.

Comment (bryceitoc9) [reply src/shared/vscode/window.ts]: Done. The newer version works as far as I can tell.

**Review by justinmk3 — APPROVED** [review]
does it need to be optional? can add that later I guess.

**Comment by bryceitoc9** [comment]
> 
> 
> does it need to be optional? can add that later I guess.

I'm eh on making it optional since that means that the constructor changes. The user can also directly call `vscode.window.withProgress`

**Pull request merged**